builders/container-apps-internal-registry-demo/startup-script.sh

```diff
@@ -28,18 +29,35 @@ if [ -n "$CORRELATION_ID" ]; then
   echo "$CORRELATION_ID" > "$build_env_dir/CORRELATION_ID"
 fi
 
-while [[ ! -f "$temp_app_source_path" || ! "$(file $temp_app_source_path)" =~ "compressed data" ]]
+content_type=""
+while [[ ! -f "$temp_app_source_path" || 
+        ! "$(file $temp_app_source_path)" =~ "compressed data" &&
+        ! $content_type =~ "application/java-archive" &&
+        ! $content_type =~ "application/gzip" &&
+        ! $content_type =~ "application/zip" ]]
 do
   echo "Waiting for app source to be uploaded. Please upload the app source to the endpoint specified in the Build resource's 'uploadEndpoint' property."
-  curl -H "$auth_header" -H "$version_header" -H "$date_header" -X GET "$file_upload_endpoint" -o "$temp_app_source_path" -s
+  curl -H "$auth_header" -H "$version_header" -H "$date_header" -X GET "$file_upload_endpoint" -o "$temp_app_source_path" -D "$temp_app_header_path" -s
   sleep 5
+ 
+  if [[ -f "$temp_app_header_path" ]]; then
+    content_type=$(grep -i Content-Type "$temp_app_header_path" | cut -d ' ' -f2)
+  else 
+    echo ""$temp_app_header_path" not found"
+  fi
```

Comment:
how about move the code to line 50? or "$temp_app_header_path not found" will be printed in console serveral times.